packages/lib/factorio/patch.js

```diff
@@ -251,8 +251,14 @@ async function patch(savePath, modules) {
 	// Write back the save
 	let tempSavePath = `${savePath}.tmp`;
 	let stream = zip.generateNodeStream({ compression: "DEFLATE" });
-	let pipe = stream.pipe(fs.createWriteStream(tempSavePath));
-	await events.once(pipe, "finish");
+	let fd = await fs.open(tempSavePath, "w");
+	try {
+		let pipe = stream.pipe(fs.createWriteStream("", { fd, autoClose: false }));
+		await events.once(pipe, "finish");
+		await fs.fsync(fd);
```

Comment:
You don't need this line - the `finally` block is always executed.

Reply:
I am confused by what you mean with this. Which line? The fsync one?

Reply:
@Hornwitser nevermind, i'm an idiot. Lack of sleep today made me thing both lines where a `.close()`